app/mongo_client.py

```diff
@@ -19,6 +19,7 @@ def get_item_from_json(self, json) -> ToDoItem:
             last_modified=json["last_modified"],
         )
 
+    @requires_role(required_role=Role.READER)
```

Comment:
We'd often put these checks in the controller (e.g. `app.py`), there can be advantages either way but this could lead to e.g. breaking transactions up if we had multiple stages in one route.